trino/client.py

```diff
@@ -607,7 +607,7 @@ def process(self, http_response) -> TrinoStatus:
             for name in get_header_values(
                 http_response.headers, constants.HEADER_DEALLOCATED_PREPARE
             ):
-                self._client_session.prepared_statements.pop(name)
+                self._client_session.prepared_statements.pop(name, None)
```

Comment:
Can we add a debug log or a `warnings.warn` here since ideally this should not be happenning at all. We should try to collect when this happens and why and fix it.

Reply:
Useful information to log would be the statement we are trying to deallocate, the current state of the prepared_statements dict and the response headers we received.